### PR TITLE
feat/#9: create module internal of Update/Upgrade

### PR DESCRIPTION
**Pull request by davidfontesfelix**
### observation: 
To run the test you need to subscribe line 19 of the internal/update/checkUpdate.go file to:
````bash
url := "https://raw.githubusercontent.com/institute-atri/wastrap/feat/%239/internal/config/config.yaml"
````
And you will also need to change the version in the internal/config/config.yaml file on line 3

# Documentation
## internal/config/config.yaml

This is where the software versioning will be located. When there is a new feature or a bug fix, the version needs to be changed.

### func checkRepositoryVersion
This function will look for config.yaml in the repository and return the version of the main branch project.

### func findConfigFile
This function will take the local config.yaml and ruturn the version

### func CheckUpdate
This function will take the return from checkRepositoryVersion and findConfigFile and compare them. If they are different, the GettingUpdate function from the internal/update/gettingUpdate.go file will be called.

### func checkTheGitPath 
Will check if project has the local git folder and will return a bool value.

### func checkIfGitIsInstalled
Will check if git is installed

### func is(OS)
Will check which operating system (OS) is installed and will return a bool value

### func openBrowser
It will open the browser on the git download website if the operating system is windows or unknown 

### func insttalingGit
This function has the methods to install git on different operating systems 

### func GettingUpdate
And finally, this function is the one that will join all the functions in the file and update the project

## Videos of the code running
[asciicast](https://asciinema.org/a/InsERe6prJ8MupZeHl8tNzChS)

### Without the git folde
[asciicast](https://asciinema.org/a/Vrl6YCxAL8jZpGHqtN6Y5tnNF)

### without the git folder and without git installed
[asciicast](https://asciinema.org/a/EUzy3NEvp1evzEtoHmDFblnQE)

### CODE REVIEW

File: internal/update/checkUpdate.go
Comment (blackcrw) [inline internal/update/checkUpdate.go:1]: The name of this file can be modified by removing the 'Update' term.

File: internal/update/gettingUpdate.go
Comment (blackcrw) [inline internal/update/gettingUpdate.go:1]: The name of this file can be modified by removing the 'Update' term.

**Comment by ygorsimoes** [comment]
Improvements you can make to make the code even better:

- Document the code - [Sequence of videos to understand godoc](https://www.youtube.com/watch?v=UnspXB5jQyI&pp=ygUNZG9jdW1lbnRhbyBnbw%3D%3D);
- Attach prints or videos of tests carried out to demonstrate how the code works - [Asciinema - Record the terminal](https://asciinema.org/);

Soon after the proposed changes we will review the PR again.

**Review by ygorsimoes — CHANGES_REQUESTED** [review]
Improvements you can make to make the code even better:

- Document the code - [Sequence of videos to understand godoc](https://www.youtube.com/watch?v=UnspXB5jQyI&pp=ygUNZG9jdW1lbnRhbyBnbw%3D%3D);
- Attach prints or videos of tests carried out to demonstrate how the code works - [Asciinema - Record the terminal](https://asciinema.org/);

Soon after the proposed changes we will review the PR again.

**Review by blackcrw — CHANGES_REQUESTED** [review]
The code is excellent, but I kindly ask you to use our standard library [gnet](https://github.com/institute-atri/gnet) for making requests and [glogger](https://github.com/institute-atri/glogger) for terminal/command prompt outputs.

The file `checkUpdate.go` is already in a folder that provides context, so it's not necessary to include `Update` in the names of the files within this folder.

I'm available to assist you with any questions regarding the use of these libraries.

**Review by blackcrw — APPROVED** [review]
LGMT

**Review by ygorsimoes — APPROVED** [review]
LGTM

**Pull request merged**